control/tests/conftest.py

```diff
@@ -26,7 +26,8 @@
                                           "PendingDeprecationWarning")
 matrixerrorfilter = pytest.mark.filterwarnings("error:.*matrix subclass:"
                                                "PendingDeprecationWarning")
-
+X11only = pytest.mark.skipif(os.getenv("DISPLAY") is None,
+                             reason="requires X11")
```

Comment:
Would not work outside of a Unix system. Is DISPLAY even set on macOS?

Reply:
It is set if you are using X11, so the mark is probably OK.  But as noted above, this may not be the actual thing to check for.  I'll see if there is a better way to trap the issue.

Reply:
This disables the tests on systems without X11 like Windows (and macOS?). And what about Linux with Wayland instead of X11?

```suggestion
hastoolbarmode = pytest.mark.skipif(
    hasattr(plt.get_current_fig_manager().toolbar, 'mode'),
    reason="incompatible matplotlib backend")
```

Reply:
I'll see if I can find another way around this, but getting rid of this causes the test to fail under GitHub actions.

Reply:
I ended up not needing the mark to get things to work (since we dropped Python 2.7), so ended up reverting this change.